vmm/src/lib.rs

```diff
@@ -919,6 +928,56 @@ impl Vmm {
                 )));
             }
 
+            // Try at most 5 passes of dirty memory sending
```

Comment:
I guess we can eventually tune that out by running more iterations and breaking when we reach a certain convergence threshold. But the code is nicely put and that can easily be added as a a follow up.

Reply:
Yeh, I think when I (and hopefully others) start migrating more complex workloads then we might need to tweak this. Fortunately the protocol allows this kind of flexibility.